### PR TITLE
Another blind attemp to work around dblclicks on Edge

### LINKED ISSUE #5180: Duplicated 'dblclick' and phantom 'touchstart' events with Chrome >= 55
Starting with chrome 55.0.2883.75 'dblclick' is fired twice

It may be related to issue #4127

Example from issue #4127 is showing the problem
http://playground-leaflet.rhcloud.com/coh/edit?html,console,output

On Firefox 50.0.2, Safari 10.0.1, Opera 41.0.2353.69, Chromium 53.0.2785.143, 
You get expected behavior:
"map click!"
"map click!"
"map double click!"
"body double click!"

While with chrome 55.0.2883.75 you get:
"map double click!"
"body double click!"
"map click!"
"map double click!"
"body double click!"

And Chrome in beta 56.0.2924.21 you get:
"map click!"
"map double click!"
"body double click!"
"map click!"
"map double click!"
"body double click!"

I still have a test machine on Chrome 54 and it has expected behavior.

### PR DESCRIPTION
**Pull request by IvanSanchez**
With a bit of luck this will fix #5180.

I've copy-pasted this patch into a playground here: https://playground-leaflet.rhcloud.com/kunur/1/edit?html,output - will appreciate testing. cc @perliedman @hyperknot @davetimmins  timlohnes 

There is also #5267, with a different approach

### CODE REVIEW

File: src/dom/DomEvent.DoubleTap.js [hunk 1]
```diff
@@ -17,9 +17,7 @@ L.extend(L.DomEvent, {
 			var count;
 
 			if (L.Browser.pointer) {
-				if ((!L.Browser.edge) || e.pointerType === 'mouse') {
```
Comment (IvanSanchez) [inline src/dom/DomEvent.DoubleTap.js]: This code won't run on anything but Edge anyway (Chrome+pointer listens only for native `dblclick`s), so the extra check is not needed.

File: src/dom/DomEvent.DoubleTap.js [hunk 2]
```diff
@@ -38,6 +36,8 @@ L.extend(L.DomEvent, {
 		function onTouchEnd() {
 			if (doubleTap && !touch.cancelBubble) {
 				if (L.Browser.pointer) {
+					if (e.pointerType === 'mouse') { return; }
```
Comment (IvanSanchez) [inline src/dom/DomEvent.DoubleTap.js]: This is why #5180 was firing `dblclick`s on mouse after a touchscreen double-tap: the `doubleTap` flag is set after a double-tap.

File: src/dom/DomEvent.DoubleTap.js [hunk 3]
```diff
@@ -65,12 +65,11 @@ L.extend(L.DomEvent, {
...
 
-		// On some platforms (notably, chrome on win10 + touchscreen + mouse),
+		// On some platforms (notably, chrome<55 on win10 + touchscreen + mouse),
 		// the browser doesn't fire touchend/pointerup events but does fire
 		// native dblclicks. See #4127.
-		if (!L.Browser.edge) {
-			obj.addEventListener('dblclick', handler, false);
-		}
+		// Edge 14 also fires native dblclicks, but only for pointerType mouse, see #5180.
+		obj.addEventListener('dblclick', handler, false);
```
Comment (IvanSanchez) [inline src/dom/DomEvent.DoubleTap.js]: This changes the strategy for Edge: always listen for native `dblblick`s, but ignore the `pointerType` mouse later.

File: src/dom/DomEvent.js [hunk 4]
```diff
@@ -77,7 +77,10 @@ L.DomEvent = {
 		if (L.Browser.pointer && type.indexOf('touch') === 0) {
 			this.addPointerListener(obj, type, handler, id);
 
-		} else if (L.Browser.touch && (type === 'dblclick') && this.addDoubleTapListener) {
+		} else if (L.Browser.touch && (type === 'dblclick') && this.addDoubleTapListener &&
+		           !(L.Browser.pointer && L.Browser.chrome) ) {
```
Comment (IvanSanchez) [inline src/dom/DomEvent.js]: This makes chrome>55 ignore the addDoubleTapListener hack and rely on native `dblclick`s.

**Comment by davetimmins** [comment]
Looks to be working as expected now 👍

**Comment by IvanSanchez** [comment]
Seems to be working as expected on:

- Edge 14 (as tested by @davetimmins and @perliedman) (native on mouse, synthetic on touch)
- Safari 9 on iPad2 (fires synthetic)
- IEMobile 11 on WinPhone 8.1 (fires native)
- Chrome 57 on Android 6 (fires native)
- Chrome 55 on desktop (fires native)
- Firefox 51 on desktop (fires native)
- Firefox 51 on Android 6 (fires synthetic)

**Comment by hyperknot** [comment]
Leaflet 1.0.2+ad27106
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36
pointerdown pointer (mouse)
mousedown mouse (1)
pointerup pointer (mouse)
mouseup mouse (1)
click mouse (1)
pointerdown pointer (mouse)
mousedown mouse (2)
pointerup pointer (mouse)
mouseup mouse (2)
click mouse (2)
**dblclick mouse (2)**

**Pull request merged**